command.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package updater
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/keybase/go-logging"
+)
+
+func timerWithTimeout(timeout time.Duration, cmd *exec.Cmd, log logging.Logger) *time.Timer {
+	return time.AfterFunc(timeout, func() {
+		if cmd != nil && cmd.Process != nil {
+			log.Warningf("Command timed out, killing")
+			if err := cmd.Process.Kill(); err != nil {
+				log.Warningf("Error trying to kill process: %s", err)
+			}
+		}
+	})
+}
+
+// RunCommand runs a command and returns the combined stdout/err output
+func RunCommand(command string, args []string, timeout time.Duration, log logging.Logger) (string, error) {
+	out, _, err := runCommand(command, args, timeout, log)
+	return out, err
+}
+
+func runCommand(command string, args []string, timeout time.Duration, log logging.Logger) (string, *os.Process, error) {
+	log.Debugf("Command: %s %s", command, args)
+	cmd := exec.Command(command, args...)
+	if cmd == nil {
+		return "", nil, fmt.Errorf("No command")
+	}
+	timer := timerWithTimeout(timeout, cmd, log)
+	if timer != nil {
+		defer timer.Stop()
+	}
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return string(out), cmd.Process, fmt.Errorf("Error running command: %s", err)
+	}
+	return string(out), cmd.Process, nil
+}
+
+// RunJSONCommand runs a command (with timeout) expecting JSON output with result interface
+func RunJSONCommand(command string, args []string, result interface{}, timeout time.Duration, log logging.Logger) error {
+	log.Debugf("Command: %s %s", command, args)
```

Comment:
I had a different impl cause I was streaming output into decoder, but we don't need that.. so I unified them, RunJSONCommand now calls runCommand.